project/Dependencies.scala

```diff
@@ -74,14 +76,26 @@ object Dependencies {
       Nil
     }
 
+  val extractVs = "[a-zA-Z]*(\\d+)\\.(\\d+)\\.(\\d+).*".r
   def sparkHive(v: String) = "org.apache.spark" %% "spark-hive" % v excludeAll(
     ExclusionRule("org.apache.hadoop"),
     ExclusionRule("org.apache.ivy", "ivy"),
     ExclusionRule("javax.servlet", "servlet-api"),
-    ExclusionRule("org.mortbay.jetty", "servlet-api"),
-    // P.S. Need to comment this out for spark 1.5.2 work with Hive+parquet in CDH 5.4.4+ (hadoop 2.6.0)    
-    ExclusionRule("com.twitter", "parquet-hadoop-bundle")
-  ) excludeAll(parquetList:_*)
+    ExclusionRule("org.mortbay.jetty", "servlet-api")
+  ) excludeAll(parquetList:_*) excludeAll(
+    {
+      val sparkVersion = defaultSparkVersion match { case extractVs(v, m, p) =>  (v.toInt, m.toInt, p.toInt)}
+      val hadoopVersion = defaultHadoopVersion match { case extractVs(v, m, p) => (v.toInt, m.toInt, p.toInt)}
+      import scala.math.Ordering.Implicits._
+      if (sparkVersion >= (1, 5, 2) && hadoopVersion >= (2, 6, 0)) {
```

Comment:
I'm afraid nothing is 100% safe with myriad of different possible configurations :(
but if we exclude `parquet-hadoop-bundle` (only) because it's included in spark, then I'd it makes sense to omit the `hadoopVersion` check.